manifests/pulp.pp

```diff
@@ -0,0 +1,135 @@
+  $db_replica_set = undef
+  $db_seeds = undef
+  $db_ssl = undef
+  $db_ssl_certfile = undef
+  $db_ssl_keyfile = undef
+  $db_unsafe_autoretry = undef
+  $db_username = undef
+  $db_verify_ssl = undef
+  $db_write_concern = undef
+  $manage_db = undef
+
+  # TODO: difference
+  if $foreman_proxy_content::pulp2_master {
+    $messaging_ca_cert      = $certs::qpid_client::qpid_client_ca_cert
+    $messaging_client_cert  = $certs::qpid_client::qpid_client_cert
+    $ca_cert                = undef
+    $https_cert             = undef
+    $https_key              = undef
+  } else {
+    $messaging_ca_cert      = pick($foreman_proxy_content::pulp_ca_cert, $certs::ca_cert)
+    $messaging_client_cert  = $certs::qpid_client_cert
+    $ca_cert                = $certs::apache::ca_cert
+    $https_cert             = $certs::apache::apache_cert
+    $https_key              = $certs::apache::apache_key
+  }
+
+  # TODO: probably useless - was passed to pulp(node) but appears unused
+  #node_server_ca_cert    => $certs::pulp_server_ca_cert,
+
+  class { 'pulp':
+    broker_url             => $broker_url,
+    broker_use_ssl         => true,
+    ca_cert                => $ca_cert,
+    db_ca_path             => $db_ca_path,
+    db_name                => $db_name,
+    db_password            => $db_password,
+    db_replica_set         => $db_replica_set,
+    db_seeds               => $db_seeds,
+    db_ssl                 => $db_ssl,
+    db_ssl_certfile        => $db_ssl_certfile,
+    db_ssl_keyfile         => $db_ssl_keyfile,
+    db_unsafe_autoretry    => $db_unsafe_autoretry,
+    db_username            => $db_username,
+    db_verify_ssl          => $db_verify_ssl,
+    db_write_concern       => $db_write_concern,
+    https_cert             => $https_cert,
+    https_key              => $https_key,
+    default_password       => $foreman_proxy_content::pulp_admin_password, # TODO: not in katello
+    enable_deb             => $foreman_proxy_content::enable_deb,
+    enable_docker          => $foreman_proxy_content::enable_docker,
+    enable_iso             => $foreman_proxy_content::enable_file,
+    enable_katello         => $enable_katello,
+    enable_ostree          => $foreman_proxy_content::enable_ostree,
+    enable_puppet          => $foreman_proxy_content::enable_puppet,
+    enable_rpm             => $foreman_proxy_content::enable_yum,
+    manage_broker          => false,
+    manage_db              => $manage_db,
+    manage_httpd           => $manage_httpd,
+    manage_plugins_httpd   => true,
+    manage_squid           => true,
+    messaging_auth_enabled => false,
+    messaging_ca_cert      => $messaging_client_cert,
+    messaging_client_cert  => $messaging_client_cert,
+    messaging_transport    => 'qpid',
+    messaging_url          => $messaging_url,
+    num_workers            => $foreman_proxy_content::pulp_num_workers,
+    proxy_password         => $foreman_proxy_content::pulp_proxy_password, # TODO: not in katello
```

Comment:
@jlsherrill  I recall you said Katello starts to configure Pulp 2 resources with explicit proxies rather than a server wide default. Does that mean we can drop proxy configs?

Note that previously only the Pulp 2 node had proxies configured where the Pulp 2 master had none. Does a node ever pull something from the internet? My impression was that they only mirror the Pulp 2 master.